src/auth/src/credentials/idtoken.rs

```diff
@@ -90,3 +96,80 @@ pub(crate) mod dynamic {
         }
     }
 }
+
+/// parse JWT ID Token string as google_cloud_auth::token::Token
+pub(crate) fn parse_id_token_from_str(token: String) -> Result<Token> {
```

Comment:
You know that `Token` is not public, right?  Even if it was: how is this supposed to be used?  What would I do with one of those `Token`s parsed from a JWT ID Token?